lib/gssapi/ntlm/init_sec_context.c

```diff
@@ -365,7 +365,7 @@ _gss_ntlm_init_sec_context
 
 	type3.username = ctx->client->username;
 	type3.flags = type2.flags;
-	type3.targetname = type2.targetname;
+	type3.targetname = name->domain;
```

Comment:
Above this are some XXX comments:

```
        /* XXX check that type2.targetinfo matches `target_name´ */
        /* XXX check verify targetinfo buffer */
```

Also, apparently the `targetname` of type 3 can be either a domain or a server name, yes?  How would we know which we should use here?